core/src/test/java/io/aiven/kafka/tieredstorage/ObjectKeyTest.java

```diff
@@ -71,6 +73,28 @@ void test() {
                     + "00000000000000001234-AAAAAAAAAAAAAAAAAAAAAA.rsm-manifest");
     }
 
+    @Test
+    void withCustomFields() {
+        final ObjectKey objectKey = new ObjectKey("test");
+        final Map<Integer, Object> fields = Map.of(
+            SegmentCustomMetadataField.OBJECT_PREFIX.index(), "other/",
+            SegmentCustomMetadataField.OBJECT_KEY.index(), "topic/7/file");
+        assertThat(objectKey.key(fields, REMOTE_LOG_SEGMENT_METADATA, ObjectKey.Suffix.LOG))
+            .isEqualTo("other/topic/7/file.log");
```

Comment:
It would be great to test this particular method with missing fields, i.e. how exactly it picks defaults.

Reply:
Good catch. Adding a couple more tests to cover default scenarios.